orttraining/orttraining/core/graph/aten_op_gradient.cc

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/core/graph/aten_op_gradient.h"
+
+#include <nlohmann/json.hpp>
+
+namespace onnxruntime {
+namespace training {
+
+using json = nlohmann::json;
+
+void ParseATenOpGradientDefinition(const std::string& grad_def_json_str, std::vector<NodeDefinition>& grad_def) {
```

Comment:
Something to consider, if we directly pybind NodeDefinition and AttributeDefinition, we won't need json serialization and parsing. 
Not a blocking comment for this PR, but let's think about the most intuitive and expressive way to define gradient in python.

Reply:
Changed to use pybind.